python/taichi/_funcs.py

```diff
@@ -9,28 +9,38 @@
 
 @func
 def _randn(dt):
-    '''
-    Generates a random number from standard normal distribution
-    using the Box-Muller transform.
-    '''
+    """
+    Generate a random float sampled from univariate standard normal
+    (Gaussian) distribution of mean 0 and variance 1, using the
+    Box-Muller transformation.
+    """
     assert dt == f32 or dt == f64
-    u1 = ops.random(dt)
+    u1 = ops.cast(1.0, dt) - ops.random(dt)
     u2 = ops.random(dt)
     r = ops.sqrt(-2 * ops.log(u1))
     c = ops.cos(math.tau * u2)
     return r * c
 
 
 def randn(dt=None):
-    """Generates a random number from standard normal distribution.
-
-    Implementation refers to :func:`taichi.lang.random.randn`.
+    """Generate a random float sampled from univariate standard normal
+    (Gaussian) distribution of mean 0 and variance 1, using the
```

Comment:
can we also explicitly list the possible range of generated number so that it's straightforward? ;)